docs/src/extend/ways-to-extend.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Ways to Extend ESLint
+eleventyNavigation:
+    key: ways to extend
+    parent: extend eslint
+    title: Ways to Extend ESLint
+    order: 1
+---
+
+ESLint is highly pluggable and configurable. There are a variety of ways that you can extend ESLint's functionality.
+
+This page explains the ways to extend ESLint, and how these extensions all fit together.
+
+## Plugins
+
+Plugins let you add your own ESLint custom rules, processors, and formatters to a project. You can publish a plugin as a npm module.
+
+Plugins are useful because your project may require some ESLint configuration that isn't included in the core `eslint` package. For example, if you're using a frontend JavaScript library like React or framework like Vue, these tools have some features that require custom rules outside the scope of the ESLint core rules.
+
+Often a plugin is paired with a shareable config, which applies a set of features from the plugin to a project. You can include a shareable config in a plugin as well.
+
+For example, [`eslint-plugin-react`](https://www.npmjs.com/package/eslint-plugin-react) is an ESLint plugin that include rules specifically for React projects. The rules include things like enforcing consistent usage of React component lifecycle methods and requiring the use of key props when rendering dynamic lists.
+
+To learn more about creating the extensions you can include in a plugin, refer to the following documentation:
+
+* [Custom Rules](custom-rules)
+* [Custom Formatters](custom-formatters)
+* [Custom Processors](custom-processors)
+* [Share Configurations](shareable-configs)
+
+To learn more about bundling these extensions into a plugin, refer to [Plugins](plugins).
```

Comment:
Formatters are not included in what we refer to as "plugins" but are separate packages.

Reply:
Also, `[Share Configurations](shareable-configs)` links to a page that explains how to create and publish a shareable config package, but that doesn't seem related to plugins. Including configurations in a plugin is different, as explained here: https://eslint.org/docs/latest/extend/plugins#configs-in-plugins

Reply:
updated!